### ISSUE TITLE
compile error on platform with usize smaller than u64

### ISSUE THREAD

**Issue by alekitto** [pavlukivan/android_trace_log#1]
Hi,
I tried to use and compile this crate against a 32bit x86 target (`i686-unknown-linux-gnu`), but the compilation fails with this message. I suspect that 32-bits ARM targets could fail in the same way.

```
error[E0277]: the trait bound `u64: ToUsize` is not satisfied
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/android_trace_log-0.2.0/src/lib.rs:685:30
    |
685 |                 input = take(event_size - bytes_read)(input)?.0;
    |                         ---- ^^^^^^^^^^^^^^^^^^^^^^^ the trait `ToUsize` is not implemented for `u64`
    |                         |
    |                         required by a bound introduced by this call
    |
    = help: the following other types implement trait `ToUsize`:
              u16
              u32
              u8
              usize
note: required by a bound in `nom::bytes::complete::take`
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/nom-6.1.2/src/bytes/complete.rs:406:6
    |
406 |   C: ToUsize,
    |      ^^^^^^^ required by this bound in `nom::bytes::complete::take`
```

**Comment by chayleaf** [comment]
thanks for reporting, should be fixed in 0.2.1

**Comment by alekitto** [comment]
I can confirm that 0.2.1 correctly compiles on i686 target.
Thank you!
